### ISSUE TITLE
Override / interpolate of variables working incorrectly

### ISSUE THREAD

**Issue by hectorvs** [theskumar/python-dotenv#256]
Bug is a combination of override and interpolate.

contents of files is as follows:

foo.txt
```
Name=foo
Interpolate="${Name}/baz"
```

bar.txt
```
Name=bar
Interpolate="${Name}/baz"
```

test file:
```
from dotenv import load_dotenv
import os

def print_env():
    for key in os.environ.keys():
       print("{}: {}".format(key, os.getenv(key)))

load_dotenv('foo.txt', override=True)
print_env()

load_dotenv('bar.txt', override=True)
print_env()
```

when second `print_env()` runs, the values are:
```
Name=bar
Interpolate=foo/baz # should be bar/baz
```

the reason for this is that the [resolve_nested_variables](https://github.com/theskumar/python-dotenv/blob/master/src/dotenv/main.py#L213) method looks into existing ENV vars first and then looks at the current file's loaded values. (in our case, `Name` was loaded in the earlier file with value `foo`). 

I believe the correct functionality would be to first look into the loaded file's values and then on the environment in order to keep the same behavior as the first file loaded.

**Comment by elbehery95** [comment]
Totally agree. This is annoying and will always enforce that the current env defined variable will still be substituted in the .env even if its still defined.

This line should just be reversed
https://github.com/theskumar/python-dotenv/blob/e92ab1f18087c2ca058203f36535c43a0f824d8a/src/dotenv/main.py#L223

**Comment by bbc2** [comment]
This indeed looks like a problem.  Thanks for reporting, I'll look into it.

**Comment by elbehery95** [comment]
I would love to have this as my first contribution in pydotenv let me know if this is possible i may submit a PR. Thanks

**Comment by bbc2** [comment]
Yes you may submit an PR.  Please try to see if the change could break an existing use case.  We might have been loading values in that order for a reason.  Let me know if you have questions about the implementation or the tests.

**Comment by bbc2** [comment]
Fixed by #258.
